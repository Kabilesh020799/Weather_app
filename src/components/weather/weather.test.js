import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Weather from './weather'; 

describe('Weather Component with Prop Data', () => {
  it('renders weather data passed via props correctly', () => {
    const weatherItem = {
      name: 'Berlin',
      main: {
        temp: 22,
        humidity: 55
      },
      weather: [{
        description: 'Clear sky'
      }],
      wind: {
        speed: 3.5
      }
    };

    render(<Weather weatherItem={weatherItem} />);

    expect(screen.getByText('Berlin')).toBeInTheDocument();
    expect(screen.getByText(/22 Â°C/)).toBeInTheDocument();
    expect(screen.getByText(/Clear sky/)).toBeInTheDocument();
    expect(screen.getByText(/55 %/)).toBeInTheDocument();
    expect(screen.getByText('3.5 m/s')).toBeInTheDocument();
  });

  it('Snapshot Testing', () => {
    const weatherItem = {
      name: 'Berlin',
      main: {
        temp: 22,
        humidity: 55
      },
      weather: [{
        description: 'Clear sky'
      }],
      wind: {
        speed: 3.5
      }
    };

    const { asFragment } = render(<Weather weatherItem={weatherItem} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
