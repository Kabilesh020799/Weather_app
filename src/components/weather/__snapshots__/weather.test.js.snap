// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="weather"
  >
    <div
      class="weather_header"
    >
      <input
        class="weather_input"
        placeholder="Enter the city name"
        value=""
      />
      <button
        class="weather_btn"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Weather Component with Prop Data Snapshot Testing 1`] = `
<DocumentFragment>
  <div
    class="weather"
  >
    <div
      class="weather_header"
    >
      <input
        class="weather_input"
        placeholder="Enter the city name"
        value=""
      />
      <button
        class="weather_btn"
      >
        Submit
      </button>
    </div>
    <div
      class="weather_body"
    >
      <div
        class="weather_body_head"
      >
        Berlin
      </div>
      <div
        class="weather_body_sub"
      >
        Current Temperature
      </div>
      <div
        class="weather_body_temp"
        id="temp"
      >
        22 Â°C 
      </div>
      <div
        class="weather_body_sub"
      >
        Weather
      </div>
      <div
        class="weather_body_temp"
        id="weather"
      >
        Clear sky
      </div>
      <div
        class="weather_body_sub"
      >
        Humidity
      </div>
      <div
        class="weather_body_temp"
        id="humidity"
      >
        55 %
      </div>
      <div
        class="weather_body_sub"
      >
        Wind Speed
      </div>
      <div
        class="weather_body_temp"
        id="wind"
      >
        3.5 m/s
      </div>
    </div>
  </div>
</DocumentFragment>
`;
